src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ILEmit/ILEmitResolverBuilder.cs

```diff
@@ -354,7 +367,7 @@ private ILEmitResolverBuilderRuntimeContext GenerateMethodBody(ServiceCallSite c
                 // Load syncLocal
                 Ldloc(context.Generator, syncLocal.LocalIndex);
                 // Load address of lockTaken
-                context.Generator.Emit(OpCodes.Ldloca_S, lockTakenLocal.LocalIndex);
+                Ldloca(context.Generator, lockTakenLocal.LocalIndex);
```

Comment:
```suggestion
                context.Generator.Emit(OpCodes.Ldloca, lockTakenLocal);
```

[`ILGenerator.Emit(OpCode, LocalInfo)`](https://github.com/dotnet/runtime/blob/1546679145d22b07eb7825eb3e16e4c61bac1994/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs#L821) already does the optimizations we do ([even on .NET Framework](https://referencesource.microsoft.com/#mscorlib/system/reflection/emit/ilgenerator.cs,861)).

Reply:
Nice catch!